ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/PushQueryResponseHandler.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.QueryResult;
+import io.confluent.ksql.api.client.Row;
+import io.confluent.ksql.api.server.protocol.QueryResponseMetadata;
+import io.confluent.ksql.rest.client.KsqlRestClientException;
+import io.vertx.core.Context;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.parsetools.RecordParser;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+
+public class PushQueryResponseHandler extends QueryResponseHandler<QueryResult> {
```

Comment:
Again, naming. This could be used for pull queries too. Maybe StreamingQueryResponseHandler would be better?

Reply:
Renamed to `StreamQueryResponseHandler`, for consistency with the methods in the `Client` interface.